### DISCUSSION TITLE
How can I build a recursive structure in TCA?

### DISCUSSION THREAD

**Discussion by tkoehlerlg** [pointfreeco/swift-composable-architecture#1914] [Q&A] [answered]
I'm wishing for an answer to enable Recursive Views Like: **View A -> View B -> View A -> ...** as well as **View A (has list of View As) -> View A (has list of View As) -> ...** and **View A (has list of View Bs) -> View B -> View A (has list of View Bs) -> ...**. I hope I made this Question not too compex but if you have questions please reach out to me. Thank you so much for answering!

[Recursive-TCA-Text.zip](https://github.com/pointfreeco/swift-composable-architecture/files/10765761/Recursive-TCA-Text.zip)

**Comment by mbrandonw** [comment]
This is possible in much the same ways it is possible in vanilla SwiftUI. What problems did you have when you attempted creating such a navigation hierarchy?

**Reply by tkoehlerlg** [reply]
Hey @mbrandonw, First thank you for your response :). At app building I receive this error: `Thread 1: EXC_BREAKPOINT (code=1, subcode=0x10c9547dc)`. On Thread 1 it fails finally on the reducer where the recusive struture loops. View A (has list of View B) -> View B -> View A <- here the View A reducer fails.

**Reply by tkoehlerlg** [reply]
May it also helps to know that the code fails with underlining `let ->viewAReducer<- = Reducer<...>`. And when I go one step back it points to: 
```
let viewBReducer<...> = .combine(
    viewAReducer.optional().pullback(...)
    Reducer { state, action, environment in
          ...
    }
)
```

**Reply by tkoehlerlg** [reply]
Okay I've reproduced it, here is the code. Just run it :)
[Recursive-TCA-Text.zip](https://github.com/pointfreeco/swift-composable-architecture/files/10765761/Recursive-TCA-Text.zip)

**Comment by tkoehlerlg** [comment, marked as answer]
Thanks to the help of the tca slack here is the solution:
https://pointfreecommunity.slack.com/files/U04PS5MQTQF/F04PWDP9P9V/recursive-tca-text-protocols.zip
